third_party/terraform/data_sources/data_source_secret_manager_secret_version.go.erb

```diff
@@ -0,0 +1,125 @@
+package google
+<% unless version == "ga" -%>
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceSecretManagerSecretVersion() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSecretManagerSecretVersionRead,
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"secret": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"destroy_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"secret_data": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func dataSourceSecretManagerSecretVersionRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	fv, err := parseProjectFieldValue("secrets", d.Get("secret").(string), "project", d, config, false)
+	if err != nil {
+		return err
+	}
+	if d.Get("project").(string) != "" && d.Get("project").(string) != fv.Project {
+		return fmt.Errorf("The project set on this secret version (%s) is not equal to the project where this secret exists (%s).", d.Get("project").(string), fv.Project)
+	}
+	project := fv.Project
+	d.Set("project", project)
+	d.Set("secret", fv.Name)
+
+	var url string
+	versionNum := d.Get("version")
+
+	if versionNum != "" {
+		url, err = replaceVars(d, config, "{{SecretManagerBasePath}}projects/{{project}}/secrets/{{secret}}/versions/{{version}}")
+		if err != nil {
+			return err
+		}
+	} else {
+		url, err = replaceVars(d, config, "{{SecretManagerBasePath}}projects/{{project}}/secrets/{{secret}}/versions/latest")
+		if err != nil {
+			return err
+		}
+	}
+
+	var version map[string]interface{}
+	version, err = sendRequest(config, "GET", project, url, nil)
+	if err != nil {
+		return fmt.Errorf("Error retrieving available secret manager secret versions: %s", err.Error())
+	}
+
+	secretVersionRegex := regexp.MustCompile("projects/(.+)/secrets/(.+)/versions/(.+)$")
+
+	parts := secretVersionRegex.FindStringSubmatch(version["name"].(string))
+	// should return [full string, project number, secret name, version number]
+	if len(parts) != 4 {
+		panic(fmt.Sprintf("secret name, %s, does not match format, projects/{{project}}/secrets/{{secret}}/versions/{{version}}", version["name"].(string)))
+	}
+
+	log.Printf("[DEBUG] Received Google SecretManager Version: %q", version)
+
+	// Get the payload
+	url = fmt.Sprintf("%s:access", url)
+	resp, err := sendRequest(config, "GET", project, url, nil)
```

Comment:
This will error if the secret version is disabled. Do we want to check the state here and only read the payload if it is enabled?

Or throw an error saying that we can't retrieve the payload if it is disabled

Reply:
Good question.  I updated it so that secret_data would just return `""` on `disabled`, but I can see both ways.
In one way, throwing an error would stay consistent with the API, but in another way (and I'm not sure of a use case for this) if the user wanted to see the other information associated with the version, it would still retrieve that information.
I could go either way.

Reply:
I'd lean towards not throwing an error if we know how to prevent it. The downside of returning `""` is pretty minimal